### PR TITLE
#11192 - Tck tests and initial support for MySQL

### PR DESCRIPTION
**Pull request by BarryDrez**
### What does this PR do?
This PR adds Tck tests for MySQL along with initial support.  Multiuser tests for MySQL to be added soon.

### What issues does this PR fix or reference?
#11192

Provide Initial support and tck tests for MySQL

#### Release Notes

Initial support for MySQL

### CODE REVIEW

**Comment by riuvshin** [comment]
Can one of the admins verify this patch?

**Comment by riuvshin** [comment]
Can one of the admins verify this patch?

**Comment by benoitf** [comment]
ci-build

**Comment by BarryDrez** [comment]
Not sure why ip-validation failed.  I have signed the ECA.  Could be the Signed-off-by footer?

**Comment by BarryDrez** [comment]
@riuvshin, I added a second commit with a sign-off footer.  I don't think I have permission to merge the commits.

**Comment by skabashnyuk** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/844/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/844//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11315
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

File: wsmaster/integration-tests/mysql-tck/mysql-support/src/main/resources/che-schema111/5.0.0-M9/mysql/2__update_local_links_in_environments.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
+--
```
Comment (mshaposhnik) [inline wsmaster/integration-tests/mysql-tck/mysql-support/src/main/resources/che-schema111/5.0.0-M9/mysql/2__update_local_links_in_environments.sql]: What is the purpose of `che-schema111`  set ?

File: core/che-core-db-vendor-mysql/pom.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>che-core-parent</artifactId>
+        <groupId>org.eclipse.che.core</groupId>
+        <version>6.12.0-SNAPSHOT</version>
+    </parent>
+    <artifactId>che-core-db-vendor-mysql</artifactId>
+    <name>Che Core :: DB :: Vendor PostgreSQL</name>
```
Comment (mshaposhnik) [inline core/che-core-db-vendor-mysql/pom.xml]: Vendor MySQL

File: wsmaster/integration-tests/mysql-tck/pom.xml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        <jdbc.url>jdbc:mysql://${docker.host.address}:${jdbc.port}/che</jdbc.url>
+        <jdbc.user.name>user</jdbc.user.name>
+        <jdbc.user.password>che1</jdbc.user.password>
+    </properties>
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>mysql</groupId>
+                <artifactId>mysql-connector-java</artifactId>
+                <version>8.0.12</version>
```
Comment (mshaposhnik) [inline wsmaster/integration-tests/mysql-tck/pom.xml:44]: That needs to be CQ-ed and moved into che-parent (we can do that btw)

File: wsmaster/integration-tests/mysql-tck/pom.xml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    <groupId>org.eclipse.che.core</groupId>
+    <artifactId>mysql</artifactId>
+    <packaging>pom</packaging>
+    <name>Integration Tests :: MySQL</name>
+    <modules>
+        <module>mysql-tck</module>
+        <module>mysql-support</module>
+    </modules>
+    <properties>
+        <db.image.name>centos/mysql-57-centos7</db.image.name>
```
Comment (mshaposhnik) [inline wsmaster/integration-tests/mysql-tck/pom.xml]: Not sure if we need that here

File: wsmaster/integration-tests/mysql-tck/pom.xml (hunk 3 above)
Comment (skabashnyuk) [reply wsmaster/integration-tests/mysql-tck/pom.xml:44]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=17680

File: core/che-core-db-vendor-mysql/src/main/java/org/eclipse/che/core/db/mysql/jpa/eclipselink/MySqlExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
```
Comment (mkuznyetsov) [inline core/che-core-db-vendor-mysql/src/main/java/org/eclipse/che/core/db/mysql/jpa/eclipselink/MySqlExceptionHandler.java]: please add `package` definition (under license header)

**Comment by skabashnyuk** [comment]
@BarryDrez can you make sure that you can compile your code with mvn clean install?

File: wsmaster/integration-tests/mysql-tck/mysql-support/src/main/resources/che-schema111/5.0.0-M9/mysql/2__update_local_links_in_environments.sql (hunk 1 above)
Comment (BarryDrez) [reply wsmaster/integration-tests/mysql-tck/mysql-support/src/main/resources/che-schema111/5.0.0-M9/mysql/2__update_local_links_in_environments.sql]: @mshaposhnik Thanks for catching this.  I meant to remove it.

**Comment by BarryDrez** [comment]
> @BarryDrez can you make sure that you can compile your code with mvn clean install?
@skabashnyuk - It has been compiling even without the import, but that may have been because I had the error handler in the same package.  I later moved it to a separate project.  I have just made some corrections based on the peer reviews, and I committed them to my fork.

**Comment by mshaposhnik** [comment]
@BarryDrez  Barry. Is it ok for you if we merge this PR into separate Che branch and finish all the required dependency stuff there (it will be faster if we do those fixes ourserves)?

**Comment by skabashnyuk** [comment]
@BarryDrez I see quite a similar modules

 wsmaster/integration-tests/mysql-tck/pom.xml
 wsmaster/integration-tests/mysql-tck/mysql-tck/pom.xml
can you take a look?

**Comment by BarryDrez** [comment]
> @BarryDrez I see quite a similar modules
> 
> wsmaster/integration-tests/mysql-tck/pom.xml
> wsmaster/integration-tests/mysql-tck/mysql-tck/pom.xml
> can you take a look?
@skabashnyuk wsmaster/integration-tests/mysql-tck/pom.xml is the parent project, and wsmaster/integration-tests/mysql-tck/mysql-tck/pom.xml runs the actual tests.
The projects have different names mysql and mysql-tck.  Is that OK, or should the directory names be different?

**Comment by BarryDrez** [comment]
> @BarryDrez Barry. Is it ok for you if we merge this PR into separate Che branch and finish all the required dependency stuff there (it will be faster if we do those fixes ourserves)?
@mshaposhnik, sure, that will be fine.  Hopefully after I get the remaining issues resolved.

**Comment by mshaposhnik** [comment]
Ok, fine.  When ready, change base or the PR to https://github.com/eclipse/che/tree/mysql_support branch and merge. We'll do the rest of dependecy stuff and merge to master than.

**Comment by BarryDrez** [comment]
@mshaposhnik, @skabashnyuk I have committed the changes as suggested.

**Pull request merged**